sdk/identity/src/token_credentials/cli_credentials.rs

```diff
@@ -18,8 +18,11 @@ mod az_cli_date_format {
         D: Deserializer<'de>,
     {
         let s = String::deserialize(deserializer)?;
-        Utc.datetime_from_str(&s, FORMAT)
-            .map_err(serde::de::Error::custom)
+        // expiresOn from azure cli uses the local timezone and needs to be converted to UTC
+        let local_datetime = Local
+            .datetime_from_str(&s, FORMAT)
+            .map_err(serde::de::Error::custom)?;
+        return Ok(local_datetime.with_timezone(&Utc));
```

Comment:
To address the issue with clippy:

```suggestion
        Ok(local_datetime.with_timezone(&Utc))
```